rust/core-lib/src/file.rs

```diff
@@ -209,6 +226,14 @@ fn try_save(path: &Path, text: &Rope, encoding: CharacterEncoding) -> io::Result
 
     fs::rename(tmp_path, path)?;
 
+    if let Some(info_unwrapped) = file_info {
+        #[cfg(target_family = "unix")]
+        fs::set_permissions(
+            path,
+            Permissions::from_mode(info_unwrapped.permissions.unwrap_or(0o644)),
```

Comment:
hmm, maybe we should be using umask to get the default?

Reply:
That does sound like a good idea, not quite sure how to get that in Rust though

Reply:
maybe we should just only set permissions when we have them stored? Otherwise we can just let rust use the defaults, which [apparently account for umask](https://doc.rust-lang.org/std/os/unix/fs/trait.OpenOptionsExt.html#tymethod.mode).

Reply:
Alright